pkg/catalogv2/git/git.go

```diff
@@ -1,480 +1,440 @@
-	}
-
-	if len(g.caBundle) > 0 {
-		f, err := ioutil.TempFile("", "ca-pem-")
+		resetOpts.Commit = commitHash
+	// FETCH_HEAD
+	default:
+		branchRef, err := r.repoGogit.Reference(plumbing.NewRemoteReferenceName("origin", reference), false)
+		// branchRef, err := ro.localRepo.Reference(plumbing.ReferenceName(reference), false)
 		if err != nil {
-			return err
-		}
-		defer os.Remove(f.Name())
-		defer f.Close()
-
-		if _, err := f.Write(g.caBundle); err != nil {
-			return fmt.Errorf("writing cabundle to %s: %w", f.Name(), err)
-		}
-		if err := f.Close(); err != nil {
-			return fmt.Errorf("closing cabundle %s: %w", f.Name(), err)
+			return fmt.Errorf("hardReset failure to get branch reference: %w", err)
 		}
-		cmd.Env = append(cmd.Env, "GIT_SSL_CAINFO="+f.Name())
+		resetOpts.Commit = branchRef.Hash()
 	}
 
-	err := cmd.Run()
+	// Validate hashCommit and reset options
+	err = resetOpts.Validate(r.repoGogit)
 	if err != nil {
-		return fmt.Errorf("git %s error: %w, detail: %v", strings.Join(args, " "), err, stderrBuf.String())
+		return fmt.Errorf("hardReset validation failure: %w", err)
 	}
-	return nil
-}
 
-func (g *git) injectAgent(cmd *exec.Cmd) (io.Closer, error) {
-	r, err := randomtoken.Generate()
+	// Open new worktree
+	wt, err := r.repoGogit.Worktree()
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("hardReset failure on WorkTree: %w", err)
 	}
 
-	tmpDir, err := ioutil.TempDir("", "ssh-agent")
+	// Reset
+	err = wt.Reset(resetOpts)
 	if err != nil {
-		return nil, err
-	}
-
-	addr := &net.UnixAddr{
-		Name: filepath.Join(tmpDir, r),
-		Net:  "unix",
+		return fmt.Errorf("hardReset failure: %w", err)
 	}
 
-	l, err := net.ListenUnix(addr.Net, addr)
-	if err != nil {
-		return nil, err
-	}
-
-	cmd.Env = append(cmd.Env, "SSH_AUTH_SOCK="+addr.Name)
+	return nil
+}
 
-	go func() {
-		defer os.RemoveAll(tmpDir)
-		defer l.Close()
-		for {
-			conn, err := l.Accept()
-			if err != nil {
-				if !k8snet.IsProbableEOF(err) {
-					logrus.Errorf("failed to accept ssh-agent client connection: %v", err)
-				}
-				return
-			}
-			if err := agent.ServeAgent(*g.agent, conn); err != nil && err != io.EOF {
-				logrus.Errorf("failed to handle ssh-agent client connection: %v", err)
-			}
-		}
-	}()
+// getLastCommitHash checks if the last commit hash for a given branch has changed in the remote repository.
+// It fetches the reference list from the remote repository and compares the commit hashes.
+//   - If the hash has not changed, it returns the same commit hash.
+//   - If it has changed, it returns the updated commit hash.
+//   - If an error occurs while fetching the reference list, an error is returned.
+func (r *repository) getLastCommitHash(branch string, commitHASH plumbing.Hash) (plumbing.Hash, error) {
 
-	return l, nil
-}
+	var lastCommitHASH plumbing.Hash
 
-func formatGitURL(endpoint, branch string) string {
-	u, err := url.Parse(endpoint)
+	remote, err := r.repoGogit.Remote(r.cloneOpts.RemoteName)
```

Comment:
Should we hardcode r.cloneOpts.RemoteName to origin ?

Reply:
Go Git already has the name "origin" as the default one.